src/components/TaskList/components/TaskItem/index.js

```diff
@@ -0,0 +1,183 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import {
+  PUBLICEYE,
+  PRIVATEEYE,
+  THUMBSUP,
+  EMPTYCIRCLE,
+  FILLEDCHECKEDCIRCLE,
+  THREEDOTS,
+} from "assets/Icon";
+import { OverlayTrigger, Popover } from "react-bootstrap";
+import Calendar from "./components/Calendar";
+
+import { updateTaskItem, selectTaskItem, deleteTaskItem } from "./actions";
+
+class TaskItem extends React.Component {
+  constructor(props) {
+    super(props);
+    const { title, dueDate } = this.props.task;
+    this.state = {
+      titleInputValue: title,
+      title,
+      dueDate,
+      isEditMode: false,
+    };
+    this.titleInput = React.createRef();
+  }
+
+  handleTitleInputChange = (event) => {
+    this.setState({ titleInputValue: event.target.value });
+  };
+
+  toggleCompletion = () => {
+    const { task, updateTask } = this.props;
+    const updatedTask = { ...task, isDone: !task.isDone };
+    updateTask(updatedTask);
+  };
+
+  togglePrivacy = () => {
+    const { task, updateTask } = this.props;
+    const updatedTask = { ...task, isPublic: !task.isPublic };
+    updateTask(updatedTask);
+  };
+
+  toggleEditMode = () => {
+    const { isEditMode } = this.state;
+    this.setState({ isEditMode: !isEditMode }, () => {
+      const updatedIsEditMode = this.state.isEditMode; 
+      if (updatedIsEditMode) {
+        this.titleInput.focus();
+      } else {
+        this.titleInput.blur();
+      }
+    });
+  };
+
+  selectTask = () => {
+    const { selectTask, task } = this.props;
+    selectTask(task);
+  };
+
+  countMangoDonations = () => {
+    const { mangoTransactions } = this.props.task;
+    return mangoTransactions.reduce((acc, curr) => {
+      return acc + curr.mangoAmount;
+    }, 0);
+  };
+
+  updateModal = () => {
+    const v = this.props.updateTasks;
+    this.props.openSubTasks();
+    return v(this.props.task);
+  };
+
+  updateTaskTitle = (event) => {
+    event.preventDefault();
+    this.toggleEditMode();
+    const { titleInputValue } = this.state;
+    const { task, updateTask } = this.props;
+    const updatedTask = {
+      ...task,
+      title: titleInputValue,
+    };
+    updateTask(updatedTask);
+  };
+
+  updateDueDate = (dueDate) => {
+    const { updateTask } = this.props;
+    const updatedTask = { ...this.props.task, dueDate };
+    updateTask(updatedTask);
+  };
+
+  deleteTask = () => {
+    const { deleteTask, task } = this.props;
```

Comment:
task unused

Reply:
Isn't "task" being passed into the deleteTask function?

Reply:
task is used to get id though?

Reply:
oh i'm blind, mb